sbi/utils/sbiutils.py

```diff
@@ -884,6 +884,63 @@ def within_support(distribution: Any, samples: Tensor) -> Tensor:
         return torch.isfinite(distribution.log_prob(samples))
 
 
+def mcmc_transform(
+    prior: Any, 
+    num_prior_samples_for_zscoring: int = 1000, 
+    enable_transform: bool = True,
+    **kwargs,
+) -> transforms.Transform:
+    """
+    Builds a transform that is applied to parameters during MCMC.
+
+    It does two things:
+    1) When the prior support is bounded, it transforms the parameters into unbounded
+        space.
+    2) It z-scores the parameters such that MCMC is performed in a z-scored space.
+
+    Args:
+        prior: The prior distribution.
+        num_prior_samples_for_zscoring: The number of samples drawn from the prior 
+            to infer the `mean` and `stddev` of the prior used for z-scoring. Unused if 
+            the prior has bounded support or when the prior has `mean` and `stddev` 
+            attributes.
+        enable_transform: Whether or not to use a transformation during MCMC.
+
+    Returns: A transformation that transforms whose `forward()` maps from unconstrained
+        (or z-scored) to constrained (or non-z-scored) space.
+    """
+
+    if enable_transform:
+        if hasattr(prior.support, "base_constraint") and hasattr(
+            prior.support.base_constraint, "upper_bound"
+        ):
+            upper = prior.support.base_constraint.upper_bound
+            lower = prior.support.base_constraint.lower_bound
+            tf = transforms.CompositeTransform(
+                [
+                    transforms.Sigmoid(),
```

Comment:
I find this quite unintuitive. We are using a Sigmoid here because effectively what the parameters will get transformed with is the inverse `logit` mapping from unconstrained to constrained space?

Reply:
Depends on what you are talking about. During `init_fn`, the parameters are indeed transformed with the `inverse` from constrained to unconstrained space.
In the potential_fn, we map the parameters from unconstrained space to constrained space. This is done via the forward mapping, i.e. `transforms.Sigmoid`